extensions/positron-python/python_files/positron/positron_ipykernel/positron_jedilsp.py

```diff
@@ -137,6 +152,50 @@ def get_message_type(self, method: str) -> Optional[Type]:
             return HelpTopicRequest
         return super().get_message_type(method)
 
+    @lsp_method(INITIALIZE)
+    def lsp_initialize(self, params: InitializeParams) -> InitializeResult:
+        result = super().lsp_initialize(params)
+
+        server = self._server
+
+        # Parse Positron-specific initialization options.
+        try:
+            raw_initialization_options = (params.initialization_options or {}).get("positron", {})
+            initialization_options = cattrs.structure(
+                raw_initialization_options, PositronInitializationOptions
+            )
+        except cattrs.BaseValidationError as error:
```

Comment:
Just curious, is there any scenario that might take this code path? I like having it for safety but I can't think of a way it might trigger

Reply:
Good question. It shouldn't be possible atm in Positron, but:

1. It was possible while I was figuring out where exactly to make the change in `positron-python` since it showed me the error in the UI. It'll be similarly helpful whenever we work on this area again.
2. It makes our language server a bit friendlier to other clients. AFAIK that isn't an explicit goal of ours, but I try to keep it in mind when it doesn't complicate things too much.